packages/backend-defaults/src/entrypoints/rootHttpRouter/rootHttpRouterServiceFactory.ts

```diff
@@ -120,7 +120,7 @@ const rootHttpRouterServiceFactoryWithOptions = (
         },
       });
 
-      lifecycle.addShutdownHook(() => server.stop());
+      lifecycle.addBeforeShutdownHook(() => server.stop());
```

Comment:
The `server.stop` needs to be part of the shutdown hook, we still want to receive traffic during the before shutdown phase. We'll want to add a separate `BeforeShutdown` hook that ensures that traffic is drained , along with the optional wait period that can be added through configuration. The traffic draining logic can be moved over from the plugin scoped HTTP router service.

Reply:
Something like this https://github.com/backstage/backstage/pull/27961/commits/cbb499d102fa4942b989fcdf6b48bfcd262ff40c?